TCAT/TCAT/HomeViewController.swift

```diff
@@ -0,0 +1,140 @@
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        navigationController?.navigationBar.barTintColor = .white
+        navigationController?.navigationBar.titleTextAttributes = [NSForegroundColorAttributeName: UIColor.white]
+        view.backgroundColor = backgroundColor
+        
+        searchBar = SearchBarView()
+        searchBar.resultsViewController?.delegate = self
+        
+        navigationItem.titleView = searchBar.searchController?.searchBar
+        
+        //UITableView Set Up
+        let tableViewFrame = CGRect(x: 0, y: searchBar.frame.maxY, width: view.bounds.width, height: view.bounds.height - searchBar.bounds.height)
+        tableView = UITableView(frame: tableViewFrame, style: .grouped)
+        tableView.backgroundColor = view.backgroundColor
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.separatorColor = tableViewSeparatorColor
+        view.addSubview(tableView)
+    }
+    
+    
+    //TableView Methods
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return retrieveRecentLocations().count == 0 ? 1 : 2
+    }
+    
+    func tableView(_ tableView: UITableView, willDisplayHeaderView view: UIView, forSection section: Int) {
+        let header = view as! UITableViewHeaderFooterView
+        header.textLabel?.textColor = headerTextColor
+        header.textLabel?.text = section == 0 ? "Cornell Destinations" : "Recent Searches"
+    }
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return section == 0 ? "Cornell Destinations" : "Recent Searches"
+    }
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return indexPath.section == 0 ? 60.0 : 50.0
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        if indexPath.section == 0 {
+            //Stub for now
+            return UITableViewCell()
+        }
+        let locations = retrieveRecentLocations()
+        let cell = tableView.dequeueReusableCell(withIdentifier: "recentLocation") == nil ? RecentSearchCell(style:UITableViewCellStyle.subtitle, reuseIdentifier: "recentLocation") : tableView.dequeueReusableCell(withIdentifier: "recentLocation")!
+        
+        cell.imageView?.frame = (frame: CGRect(x: 5, y: 5, width: 25, height: 25))
+        cell.imageView?.image = #imageLiteral(resourceName: "search")
+        cell.textLabel?.text = locations[indexPath.row].name
+        cell.preservesSuperviewLayoutMargins = false
+        cell.separatorInset = .zero
+        cell.layoutMargins = .zero
+        
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return section == 0 ? 5 : retrieveRecentLocations().count
```

Comment:
might be better to keep array of Cornell destinations and refer to that array.count